core/src/com/unciv/logic/city/CityInfo.kt

```diff
@@ -624,6 +683,42 @@ class CityInfo {
         civInfo.updateDetailedCivResources() // this building could be a resource-requiring one
     }
 
+    private fun demandNewResource() {
+        val candidates = getRuleset().tileResources.values.filter {
+            // 1. Luxury
+            it.resourceType == ResourceType.Luxury &&
+            // 2. Not a city-state only resource eg jewelry
+            !it.hasUnique(UniqueType.CityStateOnlyResource) &&
+            // 3. Not the same as last
+            it.name != demandedResource &&
+            // 4. Not found nearby
+            getCenterTile().getTilesInDistance(3).none { nearTile -> nearTile.resource == it.name }
```

Comment:
Perhaps I'm just missing it, but I don't see anywhere that checks that you don't already have the resource globally?

IIRC in Civ V it will never demand a resource you have in *any* city or trade deal, not just in your current city, unless you aleady have *every* resource, in which case it may be possible to enter a semi-permanent WLTK cycle with instant completion as soon as it's available. Wikia doesn't say much, but that's what I think I noticed from playing. (Not sure about the semi-pemanent cycle— Definitely remember it, but it may be due to leaving a single resource untapped and buying it through a trade deal each time WLTK restarted.)

It also seemed to choose the globally furthest resources, instead of just one that isn't near the affected city. More like `sorted(ruleset.resources, key=(res -> min(cities.map(city -> city.distancetoresource(res)))))[randomindexnearend]` or something than `.getTilesInDistance(3).none(resource)`. Hm. Compare this approach with Yair's comment, though.

Perhaps build a transient that maps resource types to distance to closest empire city whenever any change in empire city locations is detected? Hm. Not sure what would be best way to do that. Iterate over all cities for all resources, for `O(resourcecount*tilecount*citiecount)` performance? Check only against capital, as a faster heuristic (`O(resourcecount*mapsize)`)? Or dilate from set of all empire/city center tiles, counting how many iterations it takes to find the first copy of each resource (Faster algorithm, *except* that dilation sounds expensive).

Or leave it as it is, at least for now, either because it already behaves correctly or because it behaves incorrectly but scope creep is lethal.

Sorry for the multiple deletes/recomments. I used the wrong button to post, then highlighted the wrong lines.

Reply:
You're right, it's supposed to be a resource you *don't* have.
Distance is not a factor beyond "not within 3 tiles of this city"

Reply:
It seems that if you have every resource on the map, the WLTKD just keeps restarting with the same resource over and over.

Reply:
I probably got the impression of distance weighting just because Luxury Resources spawn regionally and I tend to gobble up any on my side of the map ASAP.

Restarting the same resource matches observed behaviour (from a long time ago), I think. IIRC for me every city would re-enter WLTK with whales at the same time whenever it ended, probably because that was the last luxury I got.